### PR TITLE
[SYCLomatic] Refined migration 8 driver APIs with bindless image.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 1]
```diff
@@ -190,11 +202,11 @@ dpct_memcpy(sycl::ext::oneapi::experimental::image_mem_handle src,
                            dest_extend, copy_extend);
 }
 
-static inline std::vector<sycl::event>
-dpct_memcpy(sycl::ext::oneapi::experimental::image_mem_handle src,
-            const sycl::ext::oneapi::experimental::image_descriptor &desc_src,
-            size_t w_offset_src, size_t h_offset_src, void *dest, size_t s,
-            sycl::queue q) {
+static inline std::vector<sycl::event> dpct_memcpy_host(
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]:
```suggestion
static inline std::vector<sycl::event> dpct_memcpy_to_host(
```

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 2]
```diff
@@ -190,11 +202,11 @@ dpct_memcpy(sycl::ext::oneapi::experimental::image_mem_handle src,
...
+    sycl::ext::oneapi::experimental::image_mem_handle src,
+    const sycl::ext::oneapi::experimental::image_descriptor &desc_src,
+    size_t w_offset_src, size_t h_offset_src, void *dest, size_t s,
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]:
```suggestion
    size_t w_offset_src, size_t h_offset_src, void *dest_host_ptr, size_t s,
```

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 3]
```diff
@@ -240,12 +272,10 @@ dpct_memcpy(const void *src,
                            dest, dest_offset, desc_dest, copy_extend);
 }
 
-static inline std::vector<sycl::event>
-dpct_memcpy(const void *src,
-            sycl::ext::oneapi::experimental::image_mem_handle dest,
-            const sycl::ext::oneapi::experimental::image_descriptor &desc_dest,
-            size_t w_offset_dest, size_t h_offset_dest, size_t s,
-            sycl::queue q = get_default_queue()) {
+static inline std::vector<sycl::event> dpct_memcpy_host(
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]:
```suggestion
static inline std::vector<sycl::event> dpct_memcpy_from_host(
```

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 4]
```diff
@@ -240,12 +272,10 @@ dpct_memcpy(const void *src,
...
+    const void *src, sycl::ext::oneapi::experimental::image_mem_handle dest,
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]:
```suggestion
    const void *src_host_ptr, sycl::ext::oneapi::experimental::image_mem_handle dest,
```

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 1 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 3 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 4 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: done

**Pull request merged**